src/components/dashboard/DashboardPage.jsx

```diff
@@ -87,14 +99,17 @@ const DashboardPage = () => {
     ),
   ];
 
+  if (!hasActiveCourseAssignments(redeemableLearnerCreditPolicies) && isFirstDashboardPageVisit()) {
+    return <EnterpriseLearnerFirstVisitRedirect />;
+  }
```

Comment:
[curious/clarification] Given  `EnterpriseLearnerFirstVisitRedirect` already has implemented checks for `isFirstDashboardPageVisit`, is this conditional rendering in this parent component largely unnecessary since `EnterpriseLearnerFirstVisitRedirect` already handles this conditional while always being rendered?

For example,  if `isFirstDashboardPageVisit()` returns false, the `isFirstDashboardPageVisit()` function called within `EnterpriseLearnerFirstVisitRedirect` would always be true.

I might recommend continuing to always render `EnterpriseLearnerFirstVisitRedirect` as before, but update the behavior of `EnterpriseLearnerFirstVisitRedirect` to take into account assignments returned by the `UserSubsidyContext`.